onnxruntime/core/providers/cuda/nn/pool.cc

```diff
@@ -251,12 +257,19 @@ Status Pool<T, MaxPool<8>, NHWC>::ComputeInternal(OpKernelContext* context) cons
   auto strides = this->pool_attrs_.strides;
 
   if (this->pool_attrs_.global_pooling) {
-    kernel_shape.assign(x_dims.begin() + 2, x_dims.end());
+    // the logic below is most likely broken. Unfortunately no test runs through this case case.
+    // accessing x_dims.end() should result in a crash since it is OOB.
+    // i assume the last element is supposed to be accessed and thus used end() -1 / end() - 2.
+    if constexpr (Layout == LAYOUT_NCHW) {
+      kernel_shape.assign(x_dims.begin() + 2, x_dims.end() - 1);
+    } else if constexpr (Layout == LAYOUT_NHWC) {
+      kernel_shape.assign(x_dims.begin() + 1, x_dims.end() - 2);
+    }
     pads.assign(kernel_shape.size(), 0);
```

Comment:
Isn't this incorrect as well? Shouldn't there be 2*kernel_shape.size() elements in the padding?